source/test/cli.ts

```diff
@@ -1,123 +1,122 @@
+test('provide a path', verifyCliFails, '', [getFixture('failure')], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('cli help flag: --help', verifyCliPasses, '', ['--help'], [
+	'Usage',
+]);
+
+test('cli version flag: --version', verifyCliPasses, '', ['--version'], [pkg.version]);
+
+test('cli typings flag: --typings', verifyCliFails, 'typings-custom-dir', ['--typings', 'utils/index.d.ts'], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('cli typings flag: -t', verifyCliFails, 'typings-custom-dir', ['-t', 'utils/index.d.ts'], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('cli files flag: --files', verifyCliFails, 'specify-test-files', ['--files', 'unknown.test.ts'], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('cli files flag: -f', verifyCliFails, 'specify-test-files', ['-f', 'unknown.test.ts'], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('cli files flag: array', verifyCliFails, 'specify-test-files', ['--files', 'unknown.test.ts', '--files', 'second.test.ts'], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('cli typings and files flags', verifyCliFails, 'typings-custom-dir', ['-t', 'utils/index.d.ts', '-f', 'index.test-d.ts'], [
+	'✖  5:19  Argument of type number is not assignable to parameter of type string.',
+	'',
+	'1 error',
+]);
+
+test('tsd logs stacktrace on failure', verifyCliFails, 'empty-package-json', [], () => {
+	const nodeModulesPath = path.resolve('node_modules');
+	const parseJsonPath = resolveFrom.silent(`${nodeModulesPath}/read-pkg`, 'parse-json') ?? `${nodeModulesPath}/index.js`;
+
+	return [
+		'Error running tsd:',
+		'JSONError: Unexpected end of JSON input while parsing empty string',
+		`at parseJson (${parseJsonPath}:29:21)`,
+		`at module.exports (${nodeModulesPath}/read-pkg/index.js:17:15)`,
+		`at async module.exports (${nodeModulesPath}/read-pkg-up/index.js:14:16)`,
+	];
+}, {startLine: 0});
 
 test('exported formatter matches cli results', async t => {
-	const options = {
-		cwd: path.join(__dirname, 'fixtures/failure'),
-	};
-
-	const {stderr: cliResults} = await t.throwsAsync<ExecaError>(execa('../../../cli.js', options));
-
-	t.true(cliResults.includes('✖  5:19  Argument of type number is not assignable to parameter of type string.'));
-
-	const tsdResults = await tsd(options);
-	const formattedResults = formatter(tsdResults);
-
-	t.true(formattedResults.includes('✖  5:19  Argument of type number is not assignable to parameter of type string.'));
+	const options = {cwd: getFixture('failure')};
+
+	const {stderr: cliResults} = await t.throwsAsync<ExecaError>(execa(binPath, options));
```

Comment:
What's the best way to test that `execa` throws while still being able to easily test the result? Since with `AVA` 4+, the result of `t.throws*` can be `undefined`.

Reply:
Either guard it or force unwrap:

```suggestion
	const {stderr: cliResults} = await t.throwsAsync<ExecaError>(execa(binPath, options))!;
```